apsuite/commisslib/topup_feedbacks.py

```diff
@@ -0,0 +1,398 @@
+        stg += ftmp('gpmodel_lengthscale', self.gpmodel_lengthscale, '[mA]')
+        stg += ftmp('gpmodel_variance', self.gpmodel_variance, '[V^2]')
+        stg += ftmp('gpmodel_noise_var', self.gpmodel_noise_var, '[V^2]')
+        stg += dtmp('gpmodel_max_num_points', self.gpmodel_max_num_points, '')
+        stg += stmp(
+            'gpmodel_bidimensional', str(self.gpmodel_bidimensional), '')
+        stg += stmp('gpmodel_issparse', str(self.gpmodel_issparse), '')
+        return stg
+
+
+class BiasFeedback(_BaseClass):
+    """."""
+
+    _DEF_TIMEOUT = 60 * 60  # [s]
+
+    def __init__(self, isonline=True, params=None):
+        """."""
+        params = params or BiasFeedbackParams()
+        super().__init__(params=params, target=self._run, isonline=isonline)
+        self._already_set = False
+
+        bias = np.linspace(
+            self.params.min_bias_voltage,
+            self.params.max_bias_voltage,
+            self.params.gpmodel_max_num_points)
+
+        b, a = self.params.coeffs_dcurr_vs_bias
+        dcurr = np.polynomial.polynomial.polyval(bias, (-b/a, 1/a))
+        y = dcurr[:, None]
+
+        if self.params.gpmodel_bidimensional:
+            self.gpmodel_kernel = gpy.kern.RBF(
+                input_dim=2, variance=self.params.gpmodel_variance,
+                lengthscale=self.params.gpmodel_lengthscale, ARD=True)
+            tim = -np.ones(bias.size, dtype=float)
+            x = np.vstack([bias, tim]).T
+        else:
+            self.gpmodel_kernel = gpy.kern.RBF(
+                input_dim=1, variance=self.params.gpmodel_variance,
+                lengthscale=self.params.gpmodel_lengthscale)
+            x = bias[:, None]
+
+        if self.params.gpmodel_issparse:
+            step = bias.size//6
+            inducing = bias[::step][:, None]
+            self.gpmodel = gpy.models.SparseGPRegression(
+                x, y, self.gpmodel_kernel, Z=inducing)
+        else:
+            self.gpmodel = gpy.models.GPRegression(
+                x, y, self.gpmodel_kernel,
+                noise_var=self.params.gpmodel_noise_var)
+
+        self.gpmodel.likelihood.variance = self.params.gpmodel_noise_var
+        self._gpmodel_pts = 0
+        if isonline:
+            _create_devices(self)
+
+    def _run(self):
+        """."""
+        print('Starting measurement:')
+        if self._stopevt.is_set():
+            return
+        egun = self.devices['egun']
+        injctrl = self.devices['injctrl']
+        currinfo = self.devices['currinfo']
+
+        self.data['dcurr'] = []
+        self.data['bias'] = []
+
+        pvo = egun.bias.pv_object('voltoutsoft')
+        pvo.auto_monitor = True
+        pvo = egun.bias.pv_object('voltinsoft')
+        pvo.auto_monitor = True
+        pvo = currinfo.si.pv_object('InjCurr-Mon')
+        pvo.auto_monitor = True
+        pvo = currinfo.bo.pv_object('Current3GeV-Mon')
+        pvo.auto_monitor = True
+        cbv = pvo.add_callback(self._callback_to_thread)
```

Comment:
maybe add a method that does this? It was done in lines 87-95 and here as well

Reply:
@murilobalves I made a lot of changes in the code, could you please take a look again?